lib/events.js

```diff
@@ -279,18 +276,14 @@ EventEmitter.prototype.removeListener =
       if (!events)
         return this;
 
-      list = events[type];
+      list = events.get(type);
       if (!list)
         return this;
 
       if (list === listener || (list.listener && list.listener === listener)) {
-        if (--this._eventsCount === 0)
-          this._events = {};
-        else {
-          delete events[type];
-          if (events.removeListener)
-            this.emit('removeListener', type, listener);
-        }
+        events.delete(type);
```

Comment:
I'm curious about performance differences with this particular change. Is it faster to keep `_eventsCount` around and create a new Map()?